src/instruments/src/PFD/HeadingIndicator.jsx

```diff
@@ -0,0 +1,137 @@
+        return null;
+    }
+
+    const bugs = [];
+
+    return (
+        <g>
+            <path id="HeadingTapeBackground" d="m32.138 145.34h73.536v10.382h-73.536z" className="TapeBackground" />
+            <HorizontalTape heading={heading} graduationElementFunction={GraduationElement} bugs={bugs} displayRange={DisplayRange + 3} valueSpacing={ValueSpacing} distanceSpacing={DistanceSpacing} />
+        </g>
+    );
+};
+
+export const HeadingOfftape = ({
+    selectedHeading, heading, ILSCourse, groundTrack,
+}) => {
+    if (getSimVar('L:A320_Neo_ADIRS_STATE', 'Enum') !== 2) {
+        return ([
+            <path id="HeadingTapeBackground" d="m32.138 145.34h73.536v10.382h-73.536z" className="TapeBackground" />,
+            <path id="HeadingTapeOutline" className="NormalStroke Red" d="m32.138 156.23v-10.886h73.536v10.886" />,
+            <text id="HDGFailText" className="Blink9Seconds FontLargest EndAlign Red" x="75.926208" y="151.95506">HDG</text>,
+        ]);
+    }
+
+    return (
+        <g id="HeadingOfftapeGroup">
+            <path id="HeadingTapeOutline" className="NormalStroke White" d="m32.138 156.23v-10.886h73.536v10.886" />
+            <SelectedHeading heading={heading} selectedHeading={selectedHeading} />
+            <QFUIndicator heading={heading} ILSCourse={ILSCourse} />
+            <path className="Fill Yellow" d="m69.61 147.31h-1.5119v-8.0635h1.5119z" />
+            <GroundTrackBug groundTrack={groundTrack} heading={heading} />
+        </g>
+    );
+};
+
+const SelectedHeading = ({ selectedHeading, heading }) => {
+    if (Number.isNaN(selectedHeading)) {
+        return null;
+    }
+
+    const headingDelta = getSmallestAngle(selectedHeading, heading);
+    const text = Math.round(selectedHeading).toString().padStart(3, '0');
+    if (Math.abs(headingDelta) < DisplayRange) {
+        const offset = headingDelta * DistanceSpacing / ValueSpacing;
+
+        return (
+            <path id="HeadingTargetIndicator" className="NormalStroke Cyan" transform={`translate(${offset} 0)`} d="m69.978 145.1 1.9501-5.3609h-6.0441l1.9501 5.3609" />
+        );
+    } if (headingDelta > 0) {
+        return (
+            <text id="SelectedHeadingTextRight" className="FontSmallest MiddleAlign Cyan" x="101.56478" y="144.44759">{text}</text>
+        );
+    }
+    return (
+        <text id="SelectedHeadingTextLeft" className="FontSmallest MiddleAlign Cyan" x="36.20676" y="144.44794">{text}</text>
+    );
+};
+
+const GroundTrackBug = ({ heading, groundTrack }) => {
+    const offset = getSmallestAngle(groundTrack, heading) * DistanceSpacing / ValueSpacing;
+    return [
+        <path className="ThickOutline" transform={`translate(${offset} 0)`} d="m68.906 145.75-1.2592 1.7639 1.2592 1.7639 1.2592-1.7639z" />,
+        <path id="ActualTrackIndicator" className="ThickStroke Green" transform={`translate(${offset} 0)`} d="m68.906 145.75-1.2592 1.7639 1.2592 1.7639 1.2592-1.7639z" />,
+    ];
```

Comment:
Same here, I recommend fragments:

```suggestion
    return <>
        <path className="ThickOutline" transform={`translate(${offset} 0)`} d="m68.906 145.75-1.2592 1.7639 1.2592 1.7639 1.2592-1.7639z" />,
        <path id="ActualTrackIndicator" className="ThickStroke Green" transform={`translate(${offset} 0)`} d="m68.906 145.75-1.2592 1.7639 1.2592 1.7639 1.2592-1.7639z" />,
    </>;
```

Reply:
Alright, I think I got all of them